src/types.ts

```diff
@@ -262,14 +263,24 @@ export interface PlpMultipleFacet extends PlpFacet {
   options: Array<PlpFacetOption>;
 }
 
-export type PlpFilterValue = string | number | boolean | Array<string | boolean | number>;
+export interface PlpSingleFacet extends PlpFacet {
+  type: 'single';
+  options: Array<PlpFacetOption>;
+}
 
+export interface PlpHierarchicalFacet extends PlpFacet {
+  type: 'hierarchical';
+  options: Array<PlpFacetOption & { options: object }>;
```

Comment:
Not sure what's the best way to do this but is there a way to make this recursive? So instead of typing the inner `options` as object, we would make it an array of PlpFacetOptions (with options added). Maybe we can create a new type `PlpHierarchicalFacetOption` and use that?

Reply:
I think it can be defined like this 
```
type PlpHierarchicalFacetOption = PlpFacetOption & { options: PlpFacetOption };
export interface PlpHierarchicalFacet extends PlpFacet {
  type: 'hierarchical';
  options: Array<PlpHierarchicalFacetOption>;
}


```